lib/features/mailbox/presentation/utils/mailbox_utils.dart

```diff
@@ -27,7 +28,12 @@ class MailboxUtils {
             ?? folderMailboxTree.findNode((node) => node.item.id == currentMailboxId);
 
         if (matchedNode != null)  {
-          final descendantIds = matchedNode.descendantsAsList().mailboxIds;
+          List<MailboxId> descendantIds;
+          if (unSubscribedChildrenMailboxIds != null) {
+            descendantIds = matchedNode.descendantsAsList().mailboxIds + unSubscribedChildrenMailboxIds;
```

Comment:
I still not get the idea here? Why add all `unSubcribed` here randomly? we can make sure it is in good order?

Reply:
This list is sorted from highest subfolder to the leaf. Take a look at the finding sub children function:
![image](https://github.com/linagora/tmail-flutter/assets/80142234/1f8346d1-a6ff-4310-801b-ead8d90bd47e)
So Given we have a folder tree like: test(test1(test2(test3)), test4(test5)).  the `unSubscribedChildrenMailboxIds` will be [test1, test2, test3, test4, test5] or [test4, test5, test1, test2, test3] (depends on test1 or test4 appears fist inside `listUnsubscribedMailboxes`).
Additionally, if this `unSubscribedChildrenMailboxIds` is not in the correct order from highest subfolder to the leaf, we will also can not fix the bug.

Reply:
We should write unit tests for this case @hieutbui @hoangdat

Reply:
no, my idea is: when we delete a mailbox, try to get all the children then delete it. WDYT?

Reply:
> no, my idea is: when we delete a mailbox, try to get all the children then delete it. WDYT?

I prefer :-p